### DISCUSSION TITLE
Is there a way to hide the app from the task list on macOS?

### DISCUSSION THREAD

**Discussion by RazrFalcon** [DeaDBeeF-Player/deadbeef#2635] [General]
### Steps to reproduce the problem

I want to be able to keep deadbeef running as a daemon on macOS without any UI. Except mpris (or whatever it's called on mac). 

### What's going on? Describe the problem in as much detail as possible.

When I'm closing/hiding deadbeef, it's still visible in Command+Tab. But there is no UI. So it simply breaks windows switching. I want to remove it from there.

I know that this is the default macOS behaviour, but maybe there is way to hide it completely and run as a daemon?

I've tried the trick with `LSUIElement`, but it still affects Command+Tab

### Information about the software:

Deadbeef version: 1.8.7-rc2 (cb5e29d) (from Homebrew)
OS: macOS 11.4

**Comment by Oleksiy-Yakovenko** [comment]
You can build deadbeef without UI, and run it from command line, but I'm not entirely sure if this is what you need.
Deadbeef doesn't really have any built in controls suitable to run it without UI full time.
There's nothing like MPRIS either.

**Reply by RazrFalcon** [reply]
But I can control it using multimedia keys and it's visible in the "media" widget on the menubar. And in the media widget I can click on the deadbeef icon and it will open the UI. So it would be great to be able to "close" UI, keeping only the daemon running.

**Reply by Oleksiy-Yakovenko** [reply]
> But I can control it using multimedia keys and it's visible in the "media" widget on the menubar. 

Deadbeef doesn't run as a daemon -- neither on linux, nor on mac.
Mac doesn't use the same principles as linux and there's nothing like MPRIS.
Media control stuff / media keys stuff is part of deadbeef UI code for mac, and won't work without the UI.
It's implemented using the following APIs:
https://developer.apple.com/documentation/mediaplayer/mpnowplayinginfocenter
https://developer.apple.com/documentation/mediaplayer/mpremotecommandcenter

**Reply by RazrFalcon** [reply]
Hmm... I see. Will see what I can do about it.

**Comment by Oleksiy-Yakovenko** [comment]
BTW I have moved it to discussions, since this doesn't seem to be a deadbeef question, but rather a general macOS UX question, that doesn't really apply to this app specifically. So perhaps in this discussion you may find your answers.

**Reply by RazrFalcon** [reply]
Sure, no problem.
